### PR TITLE
Add /msg for one-to-one messaging

### PR DESCRIPTION
**Pull request by hildjj**
/msg [nick] [message]
This is done in a way that is compatible with the architecture; it's a massive hack.

### CODE REVIEW

**Comment by martinthomson** [comment]
We probably need to have the `<br/>` addition suppressed if there is no output.  Otherwise, when others send private messages, we'll have lots of blank lines.

**Comment by bifurcation** [comment]
Seems like some different styling might be appropriate, something like:

`<i>$nick (private): $message</i>

**Comment by bifurcation** [comment]
Also, massive hacks are best hacks.

File: index.html [hunk 1]
```diff
@@ -283,6 +283,14 @@
       default: crickets.start(false); break;
     }
   },
+
+  // Direct message in the hackiest possible way: send the message
+  // to everyone, and all of the non-targets ignore it.
+  msg: function(obj) {
+    var command = obj.message.match(/^\/msg\s+(\S+)\s+(.*)/);
+    if (!command || (command[1] !== nick.value)) { return " "; }
+    return false;
```
Comment (bifurcation) [inline index.html]: This should return something styled.   Suggest:

```
return "<i>"+ nick +" (private) : "+ command[2] +"</i></br>"
```

Plus alter the regex above to also match the message.

**Comment by hildjj** [comment]
Working on the <br> thing, fixing a couple of related issues with localstorage.

File: index.html [hunk 2]
```diff
@@ -246,7 +237,8 @@
 // be run.
 //  Input:  The object received from pubnub
 //  Output: A string containing the HTML to render to the chat window,
-//     something falsy if you want the default rendering.
+//     0 to keep anything from being renndered, and something else falsy if you
```
Comment (bifurcation) [inline index.html:240]: "renndered"

File: index.html [hunk 3]
```diff
@@ -283,6 +275,21 @@
...
+    var command = obj.message.match(/^\/msg\s+(\S+)\s+(.*)/);
+    if (!command || (command[1] !== nick.value)) { return 0; }
+    return "<i>" + obj.nick + " (privately)</i>: " + command[2];
+  },
+
+  // Kick, using the totally secure approach of /msg
+  kick: function(obj) {
+    var command = obj.message.match(/^\/kick\s+(\S+)/);
+    if (!command || (command[1] !== nick.value)) { return 0; }
+    window.location.href = 'https://en.wikipedia.org/wiki/Kick';
```
Comment (bifurcation) [inline index.html:291]: Surely we can do better than this!

```
var KICK_URLs = [
  "https://en.wikipedia.org/wiki/Kick",
  "https://heavyeditorial.files.wordpress.com/2012/09/chuck-norris-super-kick.gif",
  "https://mindofburns.files.wordpress.com/2013/05/ad969-karate-kid-o.gif",
  "http://tabmok99.mortalkombatonline.com/cage1.gif",
];
window.location.href = KICK_URLs[Math.floor(Math.random() * KICK_URLs.length)]
```

**Pull request closed without merging**